app/src/main/java/com/sdp13epfl2021/projmag/database/UserProfileDatabase.kt

```diff
@@ -0,0 +1,110 @@
+package com.sdp13epfl2021.projmag.database
+
+import android.content.ContentValues.TAG
+import android.util.Log
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.ktx.toObject
+import com.sdp13epfl2021.projmag.model.*
+
+class UserProfileDatabase(private val firestore: FirebaseFirestore, private val auth: FirebaseAuth) {
+    companion object {
+        /**
+         * Root collection for user-profile
+         */
+        const val ROOT = "user-profile"
+
+        /**
+         *  The field containing favorites
+         */
+        const val FAVORITES_FIELD = "profile"
+    }
+
+    /**
+     * Return the current logged user or null if none
+     *
+     * @return current logged user or null
+     */
+    private fun getUser(): FirebaseUser? = auth.currentUser
+
+
+    /**
+     * Return the document associated to the user
+     *
+     * @return document associated to the user
+     */
+    private fun getUserDoc(): DocumentReference? =
+        getUser()?.let { user ->
+            firestore
+                .collection(UserDataFirebase.ROOT)
+                .document(user.uid)
+        }
+
+    public fun uploadProfile(profile : ImmutableProfile) {
```

Comment:
It could be usefull to have some feedback about the success/failure for both upload/get.
For example on others databse we used some callback :
```
uploadProfile(
    profile : ImmutableProfile,
    onSuccess: () -> Unit,
    onFailure: (Exception) -> Unit,
) { ... }
```

Reply:
Good suggestion will add this to the method